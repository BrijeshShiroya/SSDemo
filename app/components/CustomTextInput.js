import PropTypes from 'prop-types';
import React from 'react';
import { TextInput, View } from 'react-native';
import { Colors } from '../theme';
import styles from './styles/CustomTextInputStyles';

const CustomTextInput = (
  {
    style,
    placeholderTextColor = Colors.lightSkyBlue,
    blurOnSubmit = false,
    returnKeyType = 'next',
    ...otherProps
  },
  ref
) => (
  <View>
    <TextInput
      style={[styles.textInput, style]}
      placeholderTextColor={placeholderTextColor}
      blurOnSubmit={blurOnSubmit}
      returnKeyType={returnKeyType}
      {...otherProps}
    />
  </View>
);

CustomTextInput.propTypes = {
  style: PropTypes.object,
  placeholderTextColor: PropTypes.string,
  blurOnSubmit: PropTypes.bool,
  returnKeyType: PropTypes.string
};

export default CustomTextInput;
