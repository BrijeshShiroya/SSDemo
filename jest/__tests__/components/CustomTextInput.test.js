import 'react-native';
import React from 'react';
import CustomTextInput from '../../../app/components/CustomTextInput';

// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';

it('CustomTextInput renders correctly', () => {
  const customTextInputComponent = renderer
    .create(<CustomTextInput />)
    .toJSON();
  expect(customTextInputComponent).toMatchSnapshot();
});
