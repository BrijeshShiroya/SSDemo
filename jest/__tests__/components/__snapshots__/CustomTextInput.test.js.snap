// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CustomTextInput renders correctly 1`] = `
<View>
  <TextInput
    allowFontScaling={true}
    blurOnSubmit={false}
    rejectResponderTermination={true}
    returnKeyType="next"
    style={
      Array [
        Object {
          "marginVertical": 16.428571428571427,
        },
        undefined,
      ]
    }
    underlineColorAndroid="transparent"
  />
</View>
`;
